aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -471,10 +471,17 @@ Tensor& tril_cpu_(Tensor &self, int64_t k) {
   if (self.numel() == 0) {
     return self;
   }
-  if (!checkTrilTriuBatchContiguous(self)) self = self.contiguous();
-  AT_DISPATCH_ALL_TYPES(self.type(), "tril", [&]{
-    apply_triu_tril<scalar_t, true, false>(self, self, k);
-  });
+  if (checkTrilTriuBatchContiguous(self)) {
+    AT_DISPATCH_ALL_TYPES(self.type(), "tril", [&]{
```

Comment:
nit: I think you can avoid the calling the dispatch twice (because remember, it will expand into some large thing), by just settings result and e.g. self_contiguous using a ternary based on checkTrilTriuBatchContiguous and then copying at the end if necessary.